pkg/aws/eni/routing/routing.go

```diff
@@ -139,37 +142,38 @@ func Install(ip net.IP, info *RoutingInfo, mtu int, masq bool) error {
 // with this ID, and because this table ID equals the ENI ifindex, it's stable
 // to rely on and therefore can be reused.
 func Delete(ip net.IP) error {
-	ipWithMask := net.IPNet{
-		IP:   ip,
-		Mask: net.CIDRMask(32, 32),
-	}
+	if (option.Config.IPAM == option.IPAMENI) && (ip.To4() != nil) {
```

Comment:
if you invert this if statement this function can return earlier and we can avoid touching the remaining code of this function.

Reply:
agreed